tests/components/tado/test_service.py

```diff
@@ -0,0 +1,140 @@
+"""The serive tests for the tado platform."""
+import json
+from unittest.mock import patch
+
+import pytest
+from requests.exceptions import RequestException
+
+from homeassistant.components.tado.const import (
+    CONF_CONFIG_ENTRY,
+    CONF_READING,
+    DOMAIN,
+    SERVICE_ADD_METER_READING,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .util import async_init_integration
+
+from tests.common import MockConfigEntry, load_fixture
+
+
+async def test_has_services(
+    hass: HomeAssistant,
+) -> None:
+    """Test the existence of the Tado Service."""
+
+    await async_init_integration(hass)
+
+    assert hass.services.has_service(DOMAIN, SERVICE_ADD_METER_READING)
+
+
+async def test_add_meter_readings(
+    hass: HomeAssistant,
+) -> None:
+    """Test the add_meter_readings service."""
+
+    await async_init_integration(hass)
+
+    config_entry: MockConfigEntry = hass.config_entries.async_entries(DOMAIN)[0]
+    fixture: str = load_fixture("tado/add_readings_success.json")
+    with patch(
+        "homeassistant.components.tado.TadoConnector.set_meter_reading",
```

Comment:
We shouldn't path the internals. Patch the library instead like you do in below test

Reply:
fixed